tests/Http/Three/TagsTest.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+namespace Tests\Http\Three;
+
+use Tests\TestCase;
+use Rogue\Models\Post;
+
+class TagsTest extends TestCase
+{
+    /**
+     * Test that a POST request to /tags updates the post's tags and
+     * creates a new event and tagged entry.
+     *
+     * POST /v3/posts/:post_id/tag
+     * @return void
+     */
+    public function testTaggingAPost()
+    {
+        // Create the models that we will be using
+        $post = factory(Post::class)->create();
+
+        // Apply the tag to the post
+        $response = $this->withAdminAccessToken()->postJson('api/v3/posts/' . $post->id . '/tag', [
+            'tag_name' => 'Good Photo',
+        ]);
+
+        $response->assertStatus(200);
+
+        // Make sure that the post's tags are updated.
+        $this->assertContains('Good Photo', $post->tagNames());
+
+        // @TODO: Make sure we created a event for the tag once events are refactored.
+    }
+
+    /**
+     * Test that a non-admin cannot tag a post.
+     *
+     * POST /v3/posts/:post_id/tag
+     * @return void
+     */
+    public function testUnauthenticatedUserCannotTagAPost()
+    {
+        // Create the models that we will be using
+        $post = factory(Post::class)->create();
+
+        // Apply the tag to the post
+        $response = $this->postJson('api/v3/posts/' . $post->id . '/tag', [
+            'tag_name' => 'Good Photo',
+        ]);
+
+        $response->assertStatus(401);
+    }
+
+    /**
+     * Test that a DELETE request to /tags deletes an existing tag
+     * on a post, creates a new event, and tagged entry.
+     *
+     * DELETE /v3/posts/:post_id/tag
+     * @return void
+     */
+    public function testDeleteTagOnAPost()
+    {
+        // Create a post with a tag.
+        $post = factory(Post::class)->create();
+        $post->tag('Good Photo');
+
+        $this->assertContains('Good Photo', $post->tagNames());
+
+        $response = $this->withAdminAccessToken()->deleteJson('api/v3/posts/' . $post->id . '/tag', [
```

Comment:
for some reason, this is throwing an unauthorized 401 error and not sure why?

Reply:
@sbsmith86 I get a very looong response but seems ok? Towards the end I am seeing:
```
  #serverVariables: array:1 [
    "HTTP_AUTHORIZATION" => "Bearer {token}"
  ]
```
I also tried something like this but didn't work:
```    
$northstarId = $this->faker->northstar_id;
$response = $this->withAccessToken($northstarId, 'admin')->json('DELET', 'api/v3/posts/' . $post->id . '/tag', [
            'tag_name' => 'Good Photo',
        ]);
```

Reply:
hmmm what is the actual error message you are getting?

Reply:
```
1) Tests\Http\Three\TagsTest::testDeleteTagOnAPost
Expected status code 200 but received 401.
Failed asserting that false is true.

/home/vagrant/DoSomething.org/rogue/vendor/laravel/framework/src/Illuminate/Foundation/Testing/TestResponse.php:78
/home/vagrant/DoSomething.org/rogue/tests/Http/Three/TagsTest.php:74

2) Tests\Http\Three\TagsTest::testAddMultipleTagsAndDeleteOne
Expected status code 200 but received 401.
Failed asserting that false is true.

/home/vagrant/DoSomething.org/rogue/vendor/laravel/framework/src/Illuminate/Foundation/Testing/TestResponse.php:78
/home/vagrant/DoSomething.org/rogue/tests/Http/Three/TagsTest.php:122
```

I think it may have something to do with the `$this->middleware('auth:api');` because when I comment that out, I'm getting different errors. Digging more into this!

Reply:
yea it is something in the middleware, i think also! but it seems like this is set up similar to other routes so not sure why these tests are failing in particular. I'll punt this one to @DFurnes for authorization expertise!

Reply:
This is so strange!! Going to pull this branch down locally and dig in.

Reply:
thanks, @DFurnes !!